src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/PopupRenderer.uwp.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.ComponentModel;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Platform.UWP;
+using UWPThickness = Windows.UI.Xaml.Thickness;
+using XamlStyle = Windows.UI.Xaml.Style;
+
+[assembly: ExportRenderer(typeof(BasePopup), typeof(PopupRenderer))]
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class PopupRenderer : Flyout, IVisualElementRenderer
+	{
+		const double defaultBorderThickness = 2;
+		const double defaultSize = 600;
+		bool isDisposed = false;
+		XamlStyle flyoutStyle;
+		XamlStyle panelStyle;
+
+		public BasePopup Element { get; private set; }
+
+		internal ViewToRendererConverter.WrapperControl Control { get; private set; }
+
+		FrameworkElement IVisualElementRenderer.ContainerElement { get => null; }
+
+		VisualElement IVisualElementRenderer.Element { get => Element; }
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
+
+		public event EventHandler<PropertyChangedEventArgs> ElementPropertyChanged;
+
+		public PopupRenderer()
+		{
+		}
+
+		void IVisualElementRenderer.SetElement(VisualElement element)
+		{
+			if (element == null)
+				throw new ArgumentNullException(nameof(element));
+
+			if (!(element is BasePopup popup))
+				throw new ArgumentNullException("Element is not of type " + typeof(BasePopup), nameof(element));
+
+			var oldElement = Element;
+			Element = popup;
+			CreateControl();
+
+			Performance.Start(out var reference);
+
+			if (oldElement != null)
+				oldElement.PropertyChanged -= OnElementPropertyChanged;
+
+			element.PropertyChanged += OnElementPropertyChanged;
+
+			OnElementChanged(new ElementChangedEventArgs<BasePopup>(oldElement, Element));
+
+			Performance.Stop(reference);
+		}
+
+		void CreateControl()
+		{
+			if (Control == null)
+			{
+				Control = new Xamarin.CommunityToolkit.UI.Views.ViewToRendererConverter.WrapperControl(Element.Content);
+				Content = Control;
+			}
+		}
+
+		void InitializeStyles()
+		{
+			flyoutStyle = new XamlStyle { TargetType = typeof(FlyoutPresenter) };
+			panelStyle = new XamlStyle { TargetType = typeof(Panel) };
+		}
+
+		protected virtual void OnElementChanged(ElementChangedEventArgs<BasePopup> e)
+		{
+			if (e.NewElement != null && !isDisposed)
+			{
+				InitializeStyles();
+				SetEvents();
+				SetColor();
+				SetBorderColor();
+				SetSize();
+				SetLayout();
+				ApplyStyles();
```

Comment:
```suggestion
				ConfigureControl();
```

Reply:
I don't fully understand this suggestion, can you explain a little bit more?

Reply:
Maybe I can be wrong, because as I said I didn't run the code... From my code review, the method `ConfigureControl` call all the methods that I suggest to remove. Here is the `ConfigureControl` [implementantion](https://github.com/xamarin/XamarinCommunityToolkit/blob/4306d6efa05d7d3e98fb964ac834f1546a444dbf/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/PopupRenderer.uwp.cs#L110)

Reply:
My mistake, you are 100% correct this can be simplified. I'll update this in the next code push